packages/chats/src/BubbleMedia/MediaImage.tsx

```diff
@@ -84,14 +85,12 @@ const useImageCallback = (
 };
 
 export const MediaImage: React.FC<MediaMessage> = (message: MediaMessage) => {
-    const { onError, onLoad } = message;
+    const imageRef = React.useRef<Image>(null);
+    const { onError, onLoad, preview, data, onLayout } = message;
     const containerStyle = useBubbleContainerStyle(message);
     const bubbleBackgroundColor = useBubbleBackgroundColor(message);
     const styles = useStyles();
-    const [
-        imageOriginalSize,
-        setImageOriginalSize,
-    ] = React.useState<ImageSize | null>(null);
+    const [imageOriginalSize, setImageOriginalSize] = React.useState<ImageSize | null>(null);
```

Comment:
This was done in order not to call Image.getSize every time the screen size changes. I think this is a heavy method.
The second `useEffect` called only when data has changed - what happens not often, maybe even once.

Reply:
I'm talking about this `useEffect` https://github.com/tonlabs/UIKit/pull/1203/files#diff-17416376c351612b0b88f240da57b0844ca37235412ac0110beb380a6dcbbe0aR99-R115

Reply:
Oh, I see. Yeah, that is actually cool.
But do you think we can optimise it? Like storing it into a ref maybe, and in first useEffect react only to `maxImageSize` changes, and in `getSize` call calculations explicitly. It's just render method of the component doesn't seems to be lightweight, so IMO better to avoid unnecessary re-renders here :)